### PR TITLE
Added various test scenarios and fixed bugs that came up

### PR DESCRIPTION
**Pull request by ie208**
Added 3 client test scenarios (more should be added later)
-1- clients register, login, one sends the other a message successfully, then both unregister.
-2- clients register, one logs in and sends the other a message, the other logs in and receives the message on login successfully, then both unregister.
-3- client registers, then tries to login with an incorrect password

DbConnection: changed add/delete methods to return boolean stating whether or not transaction was successful, so that some feedback can be returned to the user. 

ServerMain: fixed various logic errors and bugs that came up from testing. Changed the map of unsent messages to use usernames as a key rather than user ID, as the user ID would require a costly database search to get (although now we run into the problem of not checking whether or not the recipient username exists at all). Also added some error handling, and improved feedback messages to clients.

### CODE REVIEW

File: ServerTesterClient/src/Client.java [hunk 1]
```diff
@@ -14,97 +14,241 @@
...
-		Client c = new Client(new URI("ws://localhost:1234"));
-		c.connect();
+
+		System.out.println("Scenario 1 - client 1 and 2 register, login, then client 1 sends client 2 a message while both are logged in.\n");
+		Client1 c1 = new Client1(new URI("ws://localhost:1234"));
+		System.out.println("Client 1 connecting");
+		c1.connect();
+		
+		try {
+			Thread.sleep(500);
```
Comment (MetallicFocus) [inline ServerTesterClient/src/Client.java]: You can use ExecutorService to wait exactly for the amount of time it needs (but less than a specific timeout). You can find the code here: #21 -- Lines 66 - 86

Comment (ie208) [reply ServerTesterClient/src/Client.java]: good point. I also want to test the order of specific things happening. Should I have an executor service for every single one of those situations?

Comment (MetallicFocus) [reply ServerTesterClient/src/Client.java]: I think you can use the same executor instance but make sure you submit a task _after_ the previous one has executed (i.e. after try-catch with timeout). When you submit a task, you can override the call (each time) to do a specific action and can attach it to the same Future instance.

Comment (MetallicFocus) [reply ServerTesterClient/src/Client.java]: As these are just used for tests, it's okay like this for now

**Pull request merged**